File: python/TEST_04_bitness.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# vim: set sw=4 ts=4 sts=4 et tw=80 :

import os, sys, unittest

sys.path.insert(0, '.')
from pinktrace import bitness, fork, trace

class TestBitness_01_Invalid(unittest.TestCase):
    def test_01_get(self):
        self.assertRaises(TypeError, bitness.get)
        self.assertRaises(TypeError, bitness.get, 'pink')

class TestBitness_02(unittest.TestCase):
    def test_02_get(self):
        pid = fork.fork()
        if (pid == 0): # child
            os._exit(13)
        else: # parent
            bit = bitness.get(pid)
            self.assertEqual(bit, bitness.DEFAULT_BITNESS)

            try: trace.kill(pid)
            except OSError: pass

if __name__ == '__main__':
    unittest.main()

File: ruby/TEST_04_bitness.rb
#!/usr/bin/env ruby
# coding: utf-8
# vim: set sw=2 sts=2 et nowrap fenc=utf-8 :
# Copyright 2010 Ali Polatel <alip@exherbo.org>
# Distributed under the terms of the GNU General Public License v2

require 'test/unit'

$: << File.expand_path('.libs')
require 'PinkTrace'

class TestPinkBitness < Test::Unit::TestCase
  def test_bitness_get_invalid
    assert_raise ArgumentError do
      PinkTrace::Bitness.get
    end
    assert_raise ArgumentError do
      PinkTrace::Bitness.get 0, 1
    end
    assert_raise TypeError do
      PinkTrace::Bitness.get 'pink'
    end
  end

  def test_bitness_get_esrch
    assert_raise Errno::ESRCH do
      PinkTrace::Bitness.get 0
    end
  end

  def test_bitness_get
    pid = PinkTrace.fork {}
    bitness = PinkTrace::Bitness.get pid
    assert(bitness == PinkTrace::Bitness::DEFAULT, "Wrong bitness, expected: #{PinkTrace::Bitness::DEFAULT} got: #{bitness}")

    begin PinkTrace::Trace.kill pid
    rescue Errno::ESRCH ;end
  end
end
